operator_csv_libs/catalog.py

```diff
@@ -148,6 +148,19 @@ def get_channels_by_substring(self, substring):
         #Return the list of channels matching the substring
         return channels_matching_substring
     
+    #Get a single channel matching a provided string, throw an exception if not found
+    def get_channel(self, channel):
+        #Find the channel, and if not found, throw an exception
+        retrieved_channel = None
+        for ch in self.channels:
+            if ch['name'] == channel:
+                retrieved_channel = ch
```

Comment:
minor change but you can add a `break` here. Shouldn't have 2 channels with the same name